src/pages/ProbeHybridisationQC.tsx

```diff
@@ -0,0 +1,391 @@
+    if (labwareForm !== undefined) {
+      if (labwareForm.completionDateTime) {
+        labwareSampleComments.completion = formatDateTimeForCore(labwareForm.completionDateTime);
+      }
+      if (labwareForm.sampleAddressComments) {
+        for (const addressSampleId in labwareForm.sampleAddressComments) {
+          const [address, sampleId] = addressSampleId.split('-');
+          labwareSampleComments.comments.push(
+            ...mapSampleAddressCommentIds(address, Number(sampleId), labwareForm.sampleAddressComments[addressSampleId])
+          );
+        }
+      }
+    }
+    completionRequest.labware.push(labwareSampleComments);
+  });
+  return completionRequest;
+};
+
+const formInitialValues: ProbeHybridisationQCFormValues = {
+  workNumber: '',
+  labwares: {
+    '': {
+      globalComments: [],
+      completionDateTime: '',
+      sampleAddressComments: { '': [] }
+    }
+  }
+};
+
+export default function ProbeHybridisationQC({ comments }: SectionComments) {
+  const stanCore = useContext(StanCoreContext);
+
+  const labwares = useCollection<LabwareFieldsFragment>({
+    getKey: (item) => item.barcode
+  });
+
+  const onAddLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.append(labware);
+    },
+    [labwares]
+  );
+
+  const onRemoveLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.remove(labware.barcode);
+    },
+    [labwares]
+  );
+
+  const removeLabwareFromForm = useCallback(
+    (removeLabware, barcode: string, values: ProbeHybridisationQCFormValues) => {
+      removeLabware(barcode);
+      delete values.labwares[barcode];
+    },
+    []
+  );
+
+  const formSubmitMachine = React.useMemo(() => {
+    return createFormMachine<CompletionRequest, RecordCompletionMutation>().withConfig({
+      services: {
+        submitForm: (ctx, e) => {
+          if (e.type !== 'SUBMIT_FORM') return Promise.reject();
+          return stanCore.RecordCompletion({
+            request: e.values
+          });
+        }
+      }
+    });
+  }, [stanCore]);
+  const [currentForm, sendForm] = useMachine(() => formSubmitMachine);
+
+  const { serverError, submissionResult } = currentForm.context;
+
+  const validationWorkNumber = Yup.object().shape({
+    workNumber: Yup.string().required('SGP Number is required')
+  });
+
+  const submitForm = async (values: CompletionRequest) => sendForm({ type: 'SUBMIT_FORM', values });
+  const convertValuesAndSubmit = async (formValues: ProbeHybridisationQCFormValues) => {
+    const values: CompletionRequest = convertProbeHybridisationQCFormValuesToCompletionRequest(
+      formValues,
+      labwares.items
+    );
+    await submitForm(values);
+  };
+
+  const mapCommentOptionsToValues = useCallback((options: OptionType[]) => {
+    return options.map((v) => v.value);
+  }, []);
+
+  const updateSectionCommentsFromGlobal = useCallback(
+    (options: OptionType[], labware: LabwareFieldsFragment, values, setFieldValue) => {
```

Comment:
It would be good to specify values ad setFieldValue as it defaults to 'any' in typescript which should generally be avoided whenever possible